Target-MVD-Mathias/ViewControllers/SignInViewController.swift

```diff
@@ -49,6 +50,44 @@ class SignInViewController: UIViewController {
     }
   }
   
+  @IBAction func facebookLogin() {
+    showSpinner()
+    let fbLoginManager = FBSDKLoginManager()
+    //Logs out before login, in case user changes facebook accounts
+    fbLoginManager.logOut()
+    fbLoginManager.logIn(withReadPermissions: ["email"], from: self) { (result, error) -> Void in
+      guard error == nil else {
+        self.showMessageError(title: "Oops..", errorMessage: "Something went wrong, try again later.")
+        self.hideSpinner()
+        return
+      }
+      if result?.grantedPermissions == nil || result?.isCancelled ?? true {
+        self.hideSpinner()
+      } else if !(result?.grantedPermissions.contains("email"))! {
+        self.hideSpinner()
+        self.showMessageError(title: "Oops..", errorMessage: "It seems that you haven't allowed Facebook to provide your email address.")
+      } else {
+        self.facebookLoginCallback()
+      }
```

Comment:
In all of this `if, else if, else` block it seems like you use that result a lot, so I would put it inside a `if let result = result {.. your code here..}` so the code looks cleaner